### PR TITLE
Misc code cleanup and organization

### PR DESCRIPTION
**Pull request by f0uriest**
- Breaks `get_fixed_boundary_constraints` etc into their own file to avoid circular imports with other utils that are used in objective classes
- Refactors bootstrap computation to avoid flake8 warning and remove use of `eval`
- Relabels some tests: `test_perturbation_orders` is now a regression test as it tends to take a long time, while a few of the example tests are now marked as unit tests since the fixture is already used by other unit tests.
- Simplify constraint projection when loading vmec solution.
- Refactors softmin/max to avoid direct JAX imports

### CODE REVIEW

File: desc/compute/_bootstrap.py [hunk 1]
```diff
@@ -189,17 +189,15 @@ def j_dot_B_Redl(
...
-            + 0.23 * (1 - 0.96 * f_t) * jnp.sqrt(nu_e) / jnp.sqrt(Zeff)
-            + 0.13
-            * (1 - 0.38 * f_t)
-            * nu_e
-            / (Zeff * Zeff)
-            * (
-                jnp.sqrt(1 + 2 * jnp.sqrt(Zeff - 1))
-                + f_t * f_t * jnp.sqrt((0.075 + 0.25 * (Zeff - 1) ** 2) * nu_e)
-            )
+        1
```
Comment (f0uriest) [inline desc/compute/_bootstrap.py:192]: This was all just autoformatting, no actual change

File: desc/objectives/__init__.py [hunk 2]
```diff
@@ -55,9 +62,3 @@
     FixThetaSFL,
 )
 from .objective_funs import ObjectiveFunction
-from .utils import (
```
Comment (ddudt) [inline desc/objectives/__init__.py:58]: Why not include the utils in the init file?

**Comment by unalmis** [comment]
can you move `def _rotation_matrix_from_normal(normal):` in `desc.compute._curve.py` to `geom_utils`. Otherwise it will get lost if you sort with https://github.com/PlasmaControl/DESC/blob/basis_vectors/docs/sort_compute_funs.py

File: desc/objectives/__init__.py (hunk 2 above)
Comment (f0uriest) [reply desc/objectives/__init__.py:58]: The stuff left in utils is not meant to be public, so its not part of the module init.

**Review by dpanici — APPROVED** [review]

**Review by ddudt — APPROVED** [review]

**Pull request merged**